src/core/net/ip6.cpp

```diff
@@ -904,6 +907,8 @@ Error Ip6::HandlePayload(Header            &aIp6Header,
         VerifyOrExit((message = aMessage.Clone()) != nullptr, error = kErrorNoBufs);
     }
 
+    VerifyOrExit(message != nullptr);
```

Comment:
I don't think we need to check this?

Based on `aMessageOwnership` we either use `&aMessage` or clone it and in the clone check we validate that it is not null.

Reply:
IIRC the `scan-build` workflow complained if we don't have this.

Reply:
Weird. Are you sure?
Why would it complain on this PR but not `main`? There seems to be no change to this method itself in this PR?

Reply:
Let's update this in a small PR of its own (not as part of this PR). I can submit a small PR to remove the check from the line ` VerifyOrExit((message = aMessage.Clone()) != nullptr, error = kErrorNoBufs);` and do it after `if` 

I guess the code checker may not be able to determine that all cases are handled based on valid `aMessageOwnership` values (or maybe they try to protect in case `aMessageOwnership happen to be a different value and none of its defined values).

Reply:
- Submitted https://github.com/openthread/openthread/pull/9462